internal/juju/secrets.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"errors"
+
+	"github.com/juju/juju/api"
+	apisecrets "github.com/juju/juju/api/client/secrets"
+	coresecrets "github.com/juju/juju/core/secrets"
+)
+
+type secretsClient struct {
+	SharedClient
+
+	getSecretAPIClient func(connection api.Connection) SecretAPIClient
+}
+
+type CreateSecretInput struct {
+	ModelName string
+	Name      string
+	Value     map[string]string
+	Info      string
+}
+
+type CreateSecretOutput struct {
+	SecretId string
+}
+
+type ReadSecretInput struct {
+	ModelName string
+	Name      string
+}
```

Comment:
You'll want to do this by SecretID and fall back to Name. 90% of our calls should use the SecretID, but for `terraform import` name will be easier for the user. 

e.g. `terraform import juju_secret.<label-name> <model-name>:<secret-name>`